data/helpers.d/php

```diff
@@ -33,23 +106,82 @@ ynh_add_fpm_config () {
 	ynh_app_setting_set --app=$app --key=phpversion --value=$phpversion
 	finalphpconf="$fpm_config_dir/pool.d/$app.conf"
 	ynh_backup_if_checksum_is_different --file="$finalphpconf"
-	cp ../conf/php-fpm.conf "$finalphpconf"
-	ynh_replace_string --match_string="__NAMETOCHANGE__" --replace_string="$app" --target_file="$finalphpconf"
-	ynh_replace_string --match_string="__FINALPATH__" --replace_string="$final_path" --target_file="$finalphpconf"
-	ynh_replace_string --match_string="__USER__" --replace_string="$app" --target_file="$finalphpconf"
-	ynh_replace_string --match_string="__PHPVERSION__" --replace_string="$phpversion" --target_file="$finalphpconf"
+
+	if [ $use_template -eq 1 ]
+	then
+                # Usage 1, use the template in ../conf/php-fpm.conf
+                cp ../conf/php-fpm.conf "$finalphpconf"
```

Comment:
Seeing your use of this helper in restore in https://github.com/YunoHost-Apps/nextcloud_ynh/pull/292, I've tried it, but with an app that uses a template (so with a `php-fpm.conf`): it fails here because in the restoration context, th file is located in `../settings/conf`.
I'm adding this comment here because I'm not sure how to address that, and that there aren't any other side effects...
So to avoid having to set a "context" (install/upgrade or restore), shall we just test for `../conf/php-fpm.conf` and fall back to `../settings/conf/php-fpm.conf` otherwise...?

Reply:
I had the same issue during restore on after implementing as done on nextcloud_ynh :
- https://github.com/YunoHost-Apps/drupal_ynh/pull/24/commits/57a2239c2c3c12f54ef55a685f5a90a2bcd152c7
- https://github.com/YunoHost-Apps/kimai2_ynh/pull/24/commits/93968e7326d7aa1d6361437ed061a5b4a0ce9829

So as a workaround, for now I switched to not using a template:
`ynh_add_fpm_config --usage=low --footprint=low --package="$extra_php_dependencies"`

Reply:
I proposed a PR to fix that: https://github.com/YunoHost/yunohost/pull/1001